log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/NamePatternConverter.java

```diff
@@ -29,6 +32,11 @@
      */
     private final NameAbbreviator abbreviator;
 
+    /**
+     * Cache of previously encountered inputs (keys) and conversion results (values).
+     */
+    private final Map<String, String> conversionCache = new ConcurrentHashMap<>();
```

Comment:
Using `CCM` for caching gives me the impression that this might have a memory impact in the long run. Though I need to dig further, I guess. @carterkozak, @jvz, @rgoers, @remkop, any ideas?

Reply:
I'm aware of applications which create an unbounded number of loggers using `LogManager.getLogger("resource." + resourceId)` to log a small amount of data. Such usage is expensive, and not good practice, but it shouldn't result in a memory leak.

I imagine if performance of abbreviation is a problem, we can improve that instead of adding caches. I'm generally a bit biased against using caches unless it's the only option, they're hard to monitor, maintain, debug, and reason about imo.

Reply:
I could modify the PR to limit the cache to, let's say, 250 entries.
Applications that 'log normally' will benefit greatly.

Reply:
I did look at improving the abbreviation, but see little room without making code even less readable (the pattern support makes for rather complicated code already).

Reply:
@carterkozak, I totally agree with your remarks on the shortcomings of introducing a caching layer. Do you have any suggestions on how to improve the performance of the existing abbreviator?

This said, @sman-81's caching hack (with a bounded size) seems like a low hanging fruit to me. We need to introduce a property where one can change the default size. Further, shall we use a `CHM` or a `ReadWriteLock`-protected `HM`?

Reply:
... or introduce a property that disables caching alltogether. I'm in favor of enabling it by default, otherwise most users will likely not benefit from it. In addition we limit the cache's size as a safeguard.

Reply:
We can use the cache size property set to `0` to disable the cache.

Reply:
Thanks for your feedback. I've added a system property to make the cache size configurable and updated the PR.

Reply:
Hello @vy, how to proceed?

Reply:
Sorry, I’ve been backlogged lately and haven’t had a chance to investigate yet

Reply:
`ping`